daemon/algod/api/server/v2/utils.go

```diff
@@ -432,7 +440,82 @@ func convertTxnResult(txnResult simulation.TxnResult) PreEncodedSimulateTxnResul
 		AppBudgetConsumed:      omitEmpty(txnResult.AppBudgetConsumed),
 		LogicSigBudgetConsumed: omitEmpty(txnResult.LogicSigBudgetConsumed),
 		TransactionTrace:       convertTxnTrace(txnResult.Trace),
+		UnnamedResources:       convertUnnamedResourceAssignment(txnResult.UnnamedResources),
+	}
+}
+
+func convertUnnamedResourceAssignment(assignment *simulation.ResourceAssignment) *model.SimulationUnnamedResourceAssignment {
+	if assignment == nil {
+		return nil
+	}
+	converted := model.SimulationUnnamedResourceAssignment{
+		MaxTotalRefs: uint64(assignment.MaxTotalRefs),
+		MaxAccounts:  uint64(assignment.MaxAccounts),
+		MaxAssets:    uint64(assignment.MaxAssets),
+		MaxApps:      uint64(assignment.MaxApps),
+		MaxBoxes:     uint64(assignment.MaxBoxes),
+		EmptyBoxRefs: omitEmpty(uint64(assignment.NumEmptyBoxRefs)),
+	}
+	if len(assignment.Accounts) != 0 {
+		accountStrings := make([]string, 0, len(assignment.Accounts))
+		for account := range assignment.Accounts {
+			accountStrings = append(accountStrings, account.String())
+		}
+		converted.Accounts = &accountStrings
+	}
+	if len(assignment.Assets) != 0 {
+		keys := uint64MapKeys(assignment.Assets)
+		converted.Assets = &keys
+	}
```

Comment:
If we had
```
func omitEmptySlice[T](slice []T) *[]T {
    // return nil if len(slice) == 0
}
```
then we could get rid of a lot of ifs, and assignments to temps.

Reply:
This reminds me of https://github.com/algorand/go-algorand/blob/master/daemon/algod/api/server/v2/utils.go#L87, which we can give it a try make it generic.

Reply:
`sliceOrNil` makes sense.

Reply:
I may have gone overboard with generics in f343aa459, but I did manage to clean up the conversion functions a lot.

Let me know what you think.